Arriba/Arriba.Client/Model/SecureDatabase.cs

```diff
@@ -51,6 +53,142 @@ public SecurityPermissions Security(string tableName)
             }
         }
 
+        public override T Query<T>(IQuery<T> query)
+        {
+            throw new ArribaException("Use Query overload which takes isCurrentUserIn on SecureDatabase.");
+        }
+
+        public T Query<T>(IQuery<T> query, Func<SecurityIdentity, bool> isCurrentUserIn)
+        {
+            ExecutionDetails preExecuteDetails = new ExecutionDetails();
+            query = SecureQuery(query, isCurrentUserIn, preExecuteDetails);
+
+            T result = base.Query<T>(query);
+            if(result is IBaseResult)
+            {
+                ((IBaseResult)result).Details.Merge(preExecuteDetails);
+            }
+
+            return result;
+        }
+
+        public IQuery<T> SecureQuery<T>(IQuery<T> query, Func<SecurityIdentity, bool> isCurrentUserIn, ExecutionDetails details)
```

Comment:
Good suggestions - I renamed, made protected, and changed the return type. These were all leftover details from my initial stab at implementing it. I kept the ExecutionDetails as an argument to be consistent with other parts of the API (and to allow a single ExecutionDetails to be reused between all of the query phases).